lib/features/permit_biometric/cubit/permit_biometric_cubit.dart

```diff
@@ -0,0 +1,94 @@
+import 'dart:io';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:givt_app/app/injection/injection.dart';
+import 'package:givt_app/core/auth/local_auth_info.dart';
+import 'package:givt_app/core/logging/logging.dart';
+import 'package:givt_app/features/permit_biometric/models/permit_biometric_request.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+part 'permit_biometric_state.dart';
+
+class PermitBiometricCubit extends Cubit<PermitBiometricState> {
+  PermitBiometricCubit({
+    required PermitBiometricRequest permitBiometricRequest,
+  }) : super(
+          PermitBiometricState.initial(
+            permitBiometricRequest: permitBiometricRequest,
+          ),
+        );
+
+  static const String _biometricsDeniedKey = 'biometricsDeniedKey';
+
+  bool get _isDenied =>
+      getIt<SharedPreferences>().getBool(_biometricsDeniedKey) ?? false;
+
+  Future<BiometricType> _getSupportedBiometric() async {
+    final isFingerprintAvailable =
+        await LocalAuthInfo.instance.checkFingerprint();
+    final isFaceIdAvailable = await LocalAuthInfo.instance.checkFaceId();
+    if (isFaceIdAvailable) {
```

Comment:
FaceId is an iOS only thing. Do you validate somewhere if thats the case? :)

Reply:
isFaceIdAvailable can be true only for iOS with FaceID I assume, so should be fine :)